### PR TITLE
Add API keys to AgenticSystemConfig instead of relying on dotenv

### PR DESCRIPTION
**Pull request by ashwinb**
Ever since we introduced distributions, "internal" tools like Brave / Wolfram were broken. This was because we got their keys from `.env`, but the CWD for starting the distribution was inside the installed `llama_toolchain` package. There's no way a user could put a `.env` there. 

Instead, I am moving the API keys specification to the `AgenticSystemConfig` -- a more natural place.

### CODE REVIEW

File: llama_toolchain/cli/download.py [hunk 1]
```diff
@@ -178,6 +178,9 @@ class ModelEntry(BaseModel):
     model_id: str
     files: Dict[str, str]
 
+    class Config:
```
Comment (hardikjshah) [inline llama_toolchain/cli/download.py:181]: why is this needed in this PR ?

**Comment by hardikjshah** [comment]
Looks good, assuming you ran test_e2e and it was all ok ?

**Review by hardikjshah — APPROVED** [review]

File: llama_toolchain/cli/download.py (hunk 1 above)
Comment (ashwinb) [reply llama_toolchain/cli/download.py:181]: it is technically not needed. it's an independent warning-fix. I can separately commit it
